src/main/java/com/minecolonies/blockout/loader/json/JsonPaneParams.java

```diff
@@ -0,0 +1,184 @@
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class JsonPaneParams implements IPaneParams
+{
+    private final JsonObject object;
+    private       View       parent;
+
+    public JsonPaneParams(final JsonObject object) {this.object = object;}
+
+    @Override
+    public String getType()
+    {
+        return object.get("type").getAsString();
+    }
+
+    @Nullable
+    @Override
+    public View getParentView()
+    {
+        return parent;
+    }
+
+    @Override
+    public void setParentView(@Nullable final View parent)
+    {
+        this.parent = parent;
+    }
+
+    @Override
+    public int getParentWidth()
+    {
+        return getParentView() != null ? getParentView().getWidth() : 0;
+    }
+
+    @Override
+    public int getParentHeight()
+    {
+        return getParentView() != null ? getParentView().getHeight() : 0;
+    }
+
+    @Nullable
+    @Override
+    public List<IPaneParams> getChildren()
+    {
+        if (!object.has("children"))
+        {
+            return ImmutableList.of();
+        }
+
+        return StreamSupport.stream(object.get("children").getAsJsonArray().spliterator(), false)
+                 .filter(JsonElement::isJsonObject)
+                 .map(JsonElement::getAsJsonObject)
+                 .map(JsonPaneParams::new)
+                 .collect(Collectors.toList());
+    }
+
+    @Nullable
+    @Override
+    public String getText()
+    {
+        return getStringAttribute("text");
+    }
+
+    @Nullable
+    @Override
+    public String getLocalizedText()
+    {
+        return Localization.localize(getText());
+    }
+
+    @Nullable
+    @Override
+    public String getStringAttribute(@NotNull final String name)
+    {
+        return getStringAttribute(name, null);
+    }
+
+    @Nullable
+    @Override
+    public String getStringAttribute(@NotNull final String name, @Nullable final String def)
+    {
+        if (!object.has(name))
+        {
+            return def;
+        }
+
+        return object.get(name).getAsString();
+    }
+
+    @Nullable
+    @Override
+    public String getLocalizedStringAttribute(@NotNull final String name)
+    {
+        return getLocalizedStringAttribute(name, null);
+    }
+
+    @Nullable
+    @Override
+    public String getLocalizedStringAttribute(@NotNull final String name, @Nullable final String def)
+    {
+        return Localization.localize(getStringAttribute(name, def));
+    }
+
+    @Override
+    public int getIntegerAttribute(@NotNull final String name)
+    {
+        return getIntegerAttribute(name, 0);
+    }
+
+    @Override
+    public int getIntegerAttribute(@NotNull final String name, final int def)
+    {
+        if (!object.has(name))
+        {
+            return def;
+        }
+
+        return object.get(name).getAsInt();
+    }
+
+    @Override
+    public float getFloatAttribute(@NotNull final String name)
+    {
+        return getFloatAttribute(name, 0f);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Upper-case this literal "f" suffix. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS818)